commit-status-publisher-server/src/main/java/jetbrains/buildServer/commitPublisher/gitlab/GitlabPublisher.java

```diff
@@ -112,9 +138,25 @@ private void publish(@NotNull String commit, @NotNull String data, @NotNull Repo
     postAsync(url, null, null, data, ContentType.APPLICATION_JSON, Collections.singletonMap("PRIVATE-TOKEN", getPrivateToken()), buildDescription);
   }
 
+  @Override
+  public void processResponse(HttpHelper.HttpResponse response) throws HttpPublisherException {
+    final int statusCode = response.getStatusCode();
+    if (statusCode >= 400) {
+      String responseString = response.getContent();
+      if (!responseString.contains("Cannot transition status via :enqueue from :pending") &&
+          !responseString.contains("Cannot transition status via :enqueue from :running") &&
+          !responseString.contains("Cannot transition status via :run from :running")) {
```

Comment:
GitLab throws a 400 error when moving between certain states, even though it actually updates the status on GitLab.

There's a closed ticket where the resolution was to basically ignore the status code:
https://gitlab.com/gitlab-org/gitlab-ce/issues/25807